### PR TITLE
Cleaning up

### PR DESCRIPTION
**Pull request by Ar3Tea**
This PR is adding:
* dynamic api call from api.js -- thanks @jpggvilaca 
* file structure changes
* cleaning up import names
* fixing prop keys in app.js to be unique

### CODE REVIEW

File: movie-browser/src/App.js [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+    isLoading: false
    };
 
   componentDidMount() {
-    fetch('https://ghibliapi.herokuapp.com/films')
-    .then(response => response.json())
-    .then(data => this.setState({ movies: data }));
-  }
-//    fetchMovies('films');
+    this.setState({ isLoading: true });
```
Comment (jpggvilaca) [inline movie-browser/src/App.js:17]: if the initial state is `isLoading: true` move that to the state itself. Remove this line.

File: movie-browser/src/App.js [hunk 2]
```diff
@@ -28,9 +29,9 @@ toggleMovieSection = () => {
     return (
       <div>
         <Header />
-        <Featured movies={movies} />
-        <MostViewed movies={movies} />
-        <Movies movies={movies} toggleMovieSection={this.toggleMovieSection} shouldShowAllMovies={this.state.shouldShowAllMovies} />
+        <Featured featuredMovies={movies} />
```
Comment (jpggvilaca) [inline movie-browser/src/App.js:32]: if `Featured` only render one movie why are u passing the whole thing?

File: movie-browser/src/App.js [hunk 3]
```diff
@@ -28,9 +29,9 @@ toggleMovieSection = () => {
...
+        <MostViewed mostViewedMovies={movies} />
```
Comment (jpggvilaca) [inline movie-browser/src/App.js:33]: same thing in here, u only need 6, not all of them

File: movie-browser/src/App.js (hunk 1 above)
Comment (Ar3Tea) [reply movie-browser/src/App.js:17]: The initial state is false. This is the starting state in app.js:
```
  state = { 
    movies: [],
    shouldShowAllMovies: false,
    isLoading: false
   };
```
The `isLoading` state only changes inside `componentDidMount` to true. And then false once the API fetch is complete.

File: movie-browser/src/App.js (hunk 2 above)
Comment (Ar3Tea) [reply movie-browser/src/App.js:32]: `Featued` renders 6 movies, but do you mean I should do this:
```
<Featured featuredMovies={movies.slice(0,6)}
```
?

File: movie-browser/src/App.js (hunk 3 above)
Comment (Ar3Tea) [reply movie-browser/src/App.js:33]: Same slice idea as I did with `Featured`?

File: movie-browser/src/App.js (hunk 2 above)
Comment (jpggvilaca) [reply movie-browser/src/App.js:32]: u should pass only the movie's it's going to render :)

File: movie-browser/src/App.js (hunk 3 above)
Comment (jpggvilaca) [reply movie-browser/src/App.js:33]: yes

**Pull request merged**